core/shared/src/main/scala/sttp/client/ResponseMetadata.scala

```diff
@@ -13,6 +13,8 @@ trait ResponseMetadata extends HasHeaders {
   def isRedirect: Boolean = code.isRedirect
   def isClientError: Boolean = code.isClientError
   def isServerError: Boolean = code.isServerError
+
+  override def toString: String = s"ResponseMetadata(code=$code, statusText=$statusText)"
```

Comment:
If we are going to omit headers totally it would be a good idea to say that explicitly to user to avoid confusion. 
```
 override def toString: String = s"ResponseMetadata(code=$code, statusText=$statusText, headers=omitted)"
```
On the other hand, we have these `HeaderNames.sensitiveHeaders`, which we might get advantage of here, but then it won't be customizable.

Reply:
Good catch, even on `Header` there is `toStringSafe` which I decided to use in the end